dask/graph_manipulation.py

```diff
@@ -21,9 +32,15 @@
 __all__ = ("bind", "checkpoint", "clone", "wait_on")
 
 T = TypeVar("T")
+try:
+    from typing import Literal  # Python >= 3.8
+
+    SplitEvery = Union[Number, Literal[False], None]
+except ImportError:
+    SplitEvery = Union[Number, bool, None]  # type: ignore
```

Comment:
The alternative to this ugly thing was to add a dependency to ``typing_extensions``. Which, if we are going to be serious with type annotations, we'll eventually have to do.

Reply:
Another alternative was to wrap the definition in ``if TYPE_CHECKING``. As long as you have a recent version of mypy,  you don't need to maintain backwards compatibility with older versions of Python.

Reply:
I think this is fine for now. Since Dask development doesn't use type annotations I'm slightly against adding infrastructure to support them.

That said, if you're passionate about type annotations then I'd encourage you to engage on https://github.com/dask/distributed/issues/2803. I recall previous discussions where type annotations were brought up with mixed reviews, but that was a while ago so it could be that people's opinions on them have changes since then.